lang/src/main/kotlin/org/partiql/lang/syntax/PartiQLVisitor.kt

```diff
@@ -99,6 +99,31 @@ internal class PartiQLVisitor(val ion: IonSystem, val customTypes: List<CustomTy
 
     override fun visitQueryDml(ctx: PartiQLParser.QueryDmlContext): PartiqlAst.PartiqlAstNode = visit(ctx.dml())
 
+    override fun visitTopStatement(ctx: PartiQLParser.TopStatementContext) = when (ctx.EXPLAIN()) {
+        null -> visit(ctx.statement()) as PartiqlAst.Statement
+        else -> PartiqlAst.build {
+            var type: String? = null
+            var format: String? = null
+            val metas = ctx.EXPLAIN().getSourceMetaContainer()
+            ctx.explainOption().forEach { option ->
+                when (option.param.text.toLowerCase()) {
+                    "type" -> { type = option.value.text }
+                    "format" -> { format = option.value.text }
+                    else -> throw option.param.error("Unknown EXPLAIN parameter.", ErrorCode.PARSE_UNEXPECTED_TOKEN)
+                }
+            }
```

Comment:
I don't believe this handles repeated params; seems like last one wins

```
EXPLAIN (TYPE AST, FORMAT DOT, TYPE LOGICAL_PLAN) ...
```

I would think this is an error. Also, do you want to normalize the value at this point with `toLowerCase()`here?

Reply:
Modified to throw error and add test.

And for normalizing, I wasn't sure whether I should, but I eventually opted not to, as the implementer may want case-sensitive values. Similar to how we received a request to remove our default population of the null spec in ORDER BY -- the requester wanted to know exactly whether the user asked for a specific null spec or not in the AST. But, in the end, it's not a big deal -- so open to modifying if there's a compelling reason.